pkg/components/references.go

```diff
@@ -21,58 +21,58 @@ import (
 	operator "github.com/tigera/operator/api/v1"
 )
 
+// variant is used to differentiate between components across product variants.
+type variant string
+
+const calicoVariant variant = "calico"
+
 type Component struct {
-	// Image is the full image path and name for this component (e.g., tigera/node, calico/cni)
-	Image   string
+	// Image is the image name for this component (e.g., node, cni)
+	Image string
+
+	// Version is the image version for this component (e.g., v3.8.1)
 	Version string
 
 	// Registry is only used for developer workflows. For production builds, the registry
 	// is always determined from user configuration. This field can be overridden
 	// as part of a developer workflow to deploy custom dev images on an individual basis.
 	Registry string
+
+	// variant is specify which product variant this component belongs to.
+	// It is used when determining default registry and image path.
+	variant variant
 }
 
 const UseDefault = "UseDefault"
 
+// getDefaults returns the default registry and imagePath for a given component.
+// This is used when no registry is explicitly defined by the component
+// and user does not explicitly specify a registry or imagePath.
+func getDefaults(c Component) (registry string, imagePath string) {
+	switch {
+	// If the component does not specify a variant, it is assumed to be an operator component.
```

Comment:
We're not really using the specific fact that the variant isn't set on the operator component here.